lib/mix/lib/mix/tasks/local.hex.ex

```diff
@@ -75,4 +85,25 @@ defmodule Mix.Tasks.Local.Hex do
         :erlang.raise(kind, reason, stacktrace)
     end
   end
+
+  defp get_matching_version do
+    Mix.Utils.read_path!(@hex_list_url)
+    |> parse_csv
+    |> calculate_version(System.version)
+  end
+
+  defp parse_csv(body) do
+    :binary.split(body, "\n", [:global, :trim])
+    |> Enum.flat_map(fn line ->
+         [_hex|elixirs] = :binary.split(line, ",", [:global, :trim])
+         elixirs
+       end)
+    |> Enum.uniq
+  end
+
+  defp calculate_version(versions, current) do
+    {:ok, current} = Version.parse(current)
+    all_eligible = Enum.filter(versions, &Version.compare(&1, current) != :gt)
```

Comment:
Isn't this going to return the Elixir version instead of the hex ones? Shouldn't it be a tuple?